src/java/io/bazel/rulesscala/scalac/CompileOptions.java

```diff
@@ -66,6 +67,7 @@ public CompileOptions(List<String> args) {
     dependencyTrackingMethod = getOrElse(argMap, "DependencyTrackingMethod", "high-level");
 
     statsfile = getOrError(argMap, "StatsfileOutput", "Missing required arg StatsfileOutput");
+    diagnosticsFile = getOrElse(argMap, "DiagnosticsFile", null);
```

Comment:
when should the else happen? Right now there is no opt-in or opt-out for this.
Relatedly-
I don't know if I feel comfortable with this being turned on by default let alone having no opt out.
Maybe this should be opt-in? Not sure how stable this reporting is or the perf impact it will have

Reply:
Sure thing! Would it be possible for you to give some pointers as to where I can find work to base myself off to implement this?

Reply:
Also, should this be a command line flag or a flag in the rules themselves?

Reply:
I'll reply to this when I have a bit more time to dig in

Reply:
Thank you for your time! I'll work on testing this then for now!

Reply:
declaration example- https://github.com/bazelbuild/rules_scala/blob/master/scala/scala_toolchain.bzl#L78
usage example- https://github.com/bazelbuild/rules_scala/blob/fdd3646bd29d42e59b4463eb975b932021da42eb/scala/private/phases/phase_dependency.bzl#L63
The example is string but you can use a bool

Reply:
Thanks for this! Let me know what you think of the implemenation.

Reply:
LGTM